ui/packages/consul-ui/app/adapters/partition.js

```diff
@@ -0,0 +1,28 @@
+import Adapter from './application';
+
+// Blocking query support for partitions is currently disabled
+export default class PartitionAdapter extends Adapter {
+  // FIXME: Check overall hierarchy again
+  async requestForQuery(request, { ns, dc, index }) {
+    const respond = await request`
+      GET /v1/partitions?${{ dc }}
+
+      ${{ index }}
+    `;
+    await respond((headers, body) => delete headers['x-consul-index']);
```

Comment:
Its just easier to do all this (re-shaping the response and the request) in the same place.

The actual request is tightly coupled to the response, so the adapter/serializer split has always felt to me like decoupling for the sake of decoupling. Doing it here for all our adapters would also mean we can get rid of a bunch of code that is a little bit out of the ordinary (our fake HTTP headers to pass this value around), plus all the tests that go with it. Ideally we'd have an even nicer format to use for re-shaping but for now this is fine.